internal/exec/exec.go

```diff
@@ -225,7 +267,38 @@ func execFieldSelection(ctx context.Context, r *Request, s *resolvable.Schema, f
 			if res.Kind() == reflect.Ptr {
 				res = res.Elem()
 			}
+			// Before hook directive visitor struct field
+			if len(f.field.Directives) > 0 {
+				for _, directive := range f.field.Directives {
+					if visitor, ok := r.Visitors[directive.Name.Name]; ok {
+						// TODO check that directive arity == 0-that should be an error at schema init time
+						visitorErr := visitor.Before(ctx, directive, nil)
+						if visitorErr != nil {
+							err := errors.Errorf("%s", visitorErr)
+							err.Path = path.toSlice()
+							err.ResolverError = visitorErr
+							return err
+						}
+					}
+				}
+			}
 			result = res.FieldByIndex(f.field.FieldIndex)
+			// After hook directive visitor (when no error is returned from resolver)
+			if !f.field.HasError && len(f.field.Directives) > 0 {
```

Comment:
I don't think that struct resolver fields can ever produce an error but I'm not an expert. Shall we delete this, Pavel?

```suggestion
			if len(f.field.Directives) > 0 {
```

Reply:
Yes, struct fields cannot produce an error. It doesn't make sense to me to check for one.